include/picongpu/fields/laserProfiles/Wavepacket.hpp

```diff
@@ -0,0 +1,270 @@
+/* Copyright 2013-2018 Axel Huebl, Heiko Burau, Rene Widera, Richard Pausch,
+ *                     Stefan Tietze
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include <pmacc/mappings/simulation/SubGrid.hpp>
+#include <pmacc/dataManagement/DataConnector.hpp>
+
+
+namespace picongpu
+{
+namespace fields
+{
+namespace laserProfiles
+{
+namespace wavepacket
+{
+    template< typename T_Params >
+    struct Unitless : public T_Params
+    {
+        using Params = T_Params;
+
+        static constexpr float_X WAVE_LENGTH = float_X( Params::WAVE_LENGTH_SI / UNIT_LENGTH ); // unit: meter
+        static constexpr float_X PULSE_LENGTH = float_X( Params::PULSE_LENGTH_SI / UNIT_TIME ); // unit: seconds (1 sigma)
+        static constexpr float_X LASER_NOFOCUS_CONSTANT = float_X( Params::LASER_NOFOCUS_CONSTANT_SI / UNIT_TIME ); //unit: seconds
+        static constexpr float_X AMPLITUDE = float_X( Params::AMPLITUDE_SI / UNIT_EFIELD ); // unit: Volt /meter
+        static constexpr float_X W0_X = float_X( Params::W0_X_SI / UNIT_LENGTH ); // unit: meter
+        static constexpr float_X W0_Z = float_X( Params::W0_Z_SI / UNIT_LENGTH ); // unit: meter
+        static constexpr float_X INIT_TIME = float_X( ( Params::PULSE_INIT * Params::PULSE_LENGTH_SI ) / UNIT_TIME ); // unit: seconds (full initialization length)
```

Comment:
This definition ignores the constant region. Therefore, the laser initialization starts in the center of the pulse. 
Old code:
```
constexpr float_X INIT_TIME = float_X ((RAMP_INIT*SI::PULSE_LENGTH_SI + SI::LASER_NOFOCUS_CONST\
ANT_SI) / UNIT_TIME); // unit: seconds (full inizialisation length)  
```
as with the plane wave laser.